visidata/apps/vdsql/_ibis.py

```diff
@@ -80,24 +81,19 @@ def open_vdsql(vd, p, filetype=None):
 class IbisConnectionPool:
     def __init__(self, source, pool=None, total=0):
         self.source = source
-        self.pool = pool if pool is not None else []
-        self.total = total
+        self._local = threading.local()
 
     def __copy__(self):
-        return IbisConnectionPool(self.source, pool=self.pool, total=self.total)
+        return IbisConnectionPool(self.source)
 
     @contextmanager
     def get_conn(self):
-        if not self.pool:
-            import ibis
-            r = ibis.connect(str(self.source))
-        else:
-            r = self.pool.pop(0)
-
+        import ibis
+        self._local.connection = getattr(self._local, 'connection', ibis.connect(str(self.source)))
```

Comment:
`ibis.connect` will be called even if the attribute exists.  So we need to do `if not threading.local().connection: threading.local().connection =  ibis.connect(...)` in order to not create extra connections only to discard them immediately.